source/MaterialXCore/Util.cpp

```diff
@@ -179,4 +181,89 @@ string parentNamePath(const string& namePath)
     return EMPTY_STRING;
 }
 
+
+std::string normalizeValueString(const std::string& str, const std::string valueType)
+{
+    // Currently supported value types.
+    StringSet supportedTypes = { "integer", "float", "vector2", "vector3", "vector4", "color3", "color4", "matrix33", "matrix44"};
+    if (supportedTypes.end() == supportedTypes.find(valueType))
+    {
+        return str;
+    }
+
+    const string TOKEN_SEPARATOR = ", ";
+
+    std::string result;
+    std::string token;
+
+    bool isInteger = (valueType == "integer");
+    if (isInteger)
+    {
+        // Remove leading and trailing spaces
+        result = str;
+        result.erase(0, result.find_first_not_of(' '));
+        result.erase(result.find_last_not_of(' ') + 1);
+    }
+    else
+    {
+        std::stringstream ss(str);
+        while (std::getline(ss, token, ','))
+        {
+            // Remove leading and trailing spaces
+            token.erase(0, token.find_first_not_of(' '));
+            token.erase(token.find_last_not_of(' ') + 1);
+
+            // Skip if the token is empty. Note that this is
+            // invalid in MaterialX but can still be stored this way. 
+            if (!token.empty())
+            {
+                // Remove leading zeros
+                token.erase(0, token.find_first_not_of('0'));
+
+                // Preserve 0 values
+                if (token.empty() || token[0] == '.')
+                {
+                    token = "0" + token;
+                }
+
+                // Check if string token has a 'e' character
+                // implying it is a floating point number in scientific notation.
+                // If it does not have an 'e' character, remove trailing zeros.
+                //
+                size_t ePos = token.find('e');
```

Comment:
This is heuristic that 'e' means there is an exponent specified.
This will "do no harm" as all that happens is that trailing 0 removal is skipped to avoid changing something like 1.2e-10 to 1.2e-1